barretenberg/cpp/src/barretenberg/dsl/acir_format/block_constraint.cpp

```
@@ -45,22 +45,22 @@ void create_block_constraints(UltraCircuitBuilder& builder,
 
     std::vector<field_ct> init;
     for (auto i : constraint.init) {
-        field_ct value = poly_to_field_ct(i, builder);
+        field_ct value = poly_triple_to_field_ct(i, builder);
         init.push_back(value);
     }
 
     switch (constraint.type) {
-    // Note: CallData/ReturnData not supported by Ultra; interpreted as ROM ops instead
-    case BlockType::CallData:
-    case BlockType::ReturnData:
-    case BlockType::ROM: {
+    case BlockType::ROM:
         process_ROM_operations(builder, constraint, has_valid_witness_assignments, init);
-    } break;
-    case BlockType::RAM: {
+        break;
+    case BlockType::RAM:
         process_RAM_operations(builder, constraint, has_valid_witness_assignments, init);
-    } break;
-    default:
-        throw_or_abort("Unexpected block constraint type.");
+        break;
+    // Note: CallData/ReturnData require DataBus, which is only available in Mega and in particular is _not_ supported
```

we should throw an error if trying to use Databus with Ultra.

---

yeah this is probably right. It might not break but it would be misleading for sure because without the inter-circuit consistency checks the databus is pretty useless